core/src/main/scala/cats/instances/seq.scala

```diff
@@ -133,6 +132,20 @@ trait SeqInstances extends cats.kernel.instances.SeqInstances {
       final override def traverse[G[_], A, B](fa: Seq[A])(f: A => G[B])(implicit G: Applicative[G]): G[Seq[B]] =
         G.map(Chain.traverseViaChain(fa.toIndexedSeq)(f))(_.toVector)
 
+      override def mapAccumulate[S, A, B](init: S, fa: Seq[A])(f: (S, A) => (S, B)): (S, Seq[B]) = {
+        var state = init
+
+        val fb = fa.iterator.map { a =>
+          val (newState, b) = f(state, a)
+
+          state = newState
+
+          b
+        }.toIndexedSeq
```

Comment:
The same comment as for the `NonEmptySeq` above.

Actually, I think that the only implementation should be put here and then re-used from those for `Traverse[NonEmptySeq]`.

Reply:
Also I still think that it would be better to preserve the original Seq's concrete type. The only thing, it should be a bit different for Scala 2.12 and 2.13+:
```
val builder = fa.companion.newBuilder[B] // 2.12
OR
val builder = fa.iterableFactory.newBuilder[B] // 2.13+
```
[Seq#companion](https://www.scala-lang.org/api/2.12.15/scala/collection/immutable/Seq.html#companion:scala.collection.generic.GenericCompanion[scala.collection.immutable.Seq])
[Seq#iterableFactory](https://www.scala-lang.org/api/current/scala/collection/immutable/Seq.html#iterableFactory:scala.collection.SeqFactory[Seq])